src/filelock/_api.py

```diff
@@ -76,12 +77,41 @@ class ThreadLocalFileContext(FileLockContext, local):
 class BaseFileLock(ABC, contextlib.ContextDecorator):
     """Abstract base class for a file lock object."""
 
-    def __init__(
+    _instances: ClassVar[WeakValueDictionary[str, BaseFileLock]] = WeakValueDictionary()
+
+    def __new__(  # noqa: PYI034
+        cls,
+        lock_file: str | os.PathLike[str],
+        is_singleton: bool = False,  # noqa: FBT001, FBT002
+        *args: Sequence[Any],  # noqa: ARG003
+        **kwargs: Mapping[str, Any],  # noqa: ARG003
```

Comment:
I don't think you need args/kwargs here 🤔

Reply:
The signature of `__new__` should match that of `__init__` or handle the extra args/kwargs as is done here, otherwise those options will not be passed on to `__init__`.

Ref: https://docs.python.org/3/reference/datamodel.html#classes.